cclib/parser/gamessdatparser.py

```diff
@@ -151,96 +156,192 @@ def extract(self, inputfile, line):
+
+        if line[0:8] == "GAUSSIAN":
+
+            parts = line.split()
+
+            self.nmo    = int(parts[1])
+            self.nbasis = int(parts[4])
+            self.natom  = int(parts[6])
+
+
+            # Continue extracting
             
-            while 'MOMENTS AT POINT' not in line:
-                fields = line.split()
-                atom_info = {
-                    "atom": fields[0],
-                    "charge": float(fields[1]),
-                    "spin": float(fields[2]),
-                    "net_charge": float(fields[3]),
-                    "net_spin": float(fields[4])
-                }
-                self.metadata["population"].append(atom_info)
-                line = next(inputfile).strip()
-        
+            #   O    1    (CENTRE  1)   0.00000000  0.00000000  0.00000000  CHARGE =  8.0
+            #   H    2    (CENTRE  2)   1.87082873  0.00000000  0.00000000  CHARGE =  1.0
+            #   H    3    (CENTRE  3)  -0.51567032  1.79835585  0.00000000  CHARGE =  1.0
 
-        # Extracting Moments at Point
+            line = next(inputfile)
 
-        # MOMENTS AT POINT    1 X,Y,Z=  0.075831  0.100631  0.000000
-        # MP2 NATURAL ORBITALS, E(MP2)=      -75.0022821133
+            atom_info = []
+            self.atomcharges = dict()
+            self.atomcharges['CHANGENAME'] = []
+            self.atomnos = []
+            
+            while '(CENTRE' in line:
 
-        if line[1:17] == "MOMENTS AT POINT":
-            moment_line = line.split("=")[1].strip()
-            moment_values = moment_line.split()[1:]
-            self.moments = [float(value) for value in moment_values]
+                parts = line.split()
 
+                symbol = parts[0]
+                atomno = self.pt.number[symbol]
 
-        # Extracting Dipole
+                # atom_number = int(parts[1])
+                # centre_number = int(parts[3][:-1])
+                val1 = float(parts[4])
+                val2 = float(parts[5])
+                val3 = float(parts[6])
 
-        # DIPOLE       1.007144  1.336525  0.000000
+                charge = float(parts[-1])
 
-        if line[1:7] == "DIPOLE":
-            line = next(inputfile).strip()
-            dipole_pattern = r"DIPOLE\s+([\d.-]+)\s+([\d.-]+)\s+([\d.-]+)"
-            match = re.search(dipole_pattern, line)
-            if match:
-                self.metadata["dipole"] = {
-                    "x": float(match.group(1)),
-                    "y": float(match.group(2)),
-                    "z": float(match.group(3))
-                }
+                atom_info.append((val1, val2, val3))
                 
-        
-        # Extracting MP2 Energy Value
-
-        # MP2 NATURAL ORBITALS, E(MP2)=      -75.0022821133
-        
-        if "E(MP2)=" in line:
-            self.mpenergies = float(line.split()[-1])
+                self.atomnos.append(atomno)
+                self.atomcharges['CHANGENAME'].append(charge)
```

Comment:
What is `CHANGENAME`?

Reply:
I wasn't sure what key name I should choose, as there are multiple, and I'm not sure on the differences.
https://cclib.github.io/data_notes.html#atomcharges

Reply:
I understand now; this is a little confusing.  `atomcharges` is the result of an _output_: those atomic charge schemes are a calculation done for an analysis, and each one is an approximation for what the partial charge on each atom might be.

This charge column is an *input* for what the nuclear charge is entering the calculation and is almost always going to be the atomic number.

We have to be careful though, because that doesn't mean it's `atomnos`.  You could modify the charge of, say, oxygen, to be 3 going into the calculation, but its atomic number is still 8.  `atomnos` in this case needs to be calculated from the atomic symbol and `PeriodicTable`.